### ISSUE TITLE
Explorer sidebar opens on new file if activity bar is hidden, after latest update

### ISSUE THREAD

**Issue by Yoraiz0r** [microsoft/vscode#230575]
Type: <b>Bug</b>

I've had VSCode set to a rather minimal layout with only text mostly, for over a year.
After the latest update, whenever I open it, the 'Explorer' opens on the sidebar.
I've reset my settings and the issue persisted, so I began adding to the settings.json one by one, until I've hit the offending setting.

It seems to be `"workbench.activityBar.location": "hidden",`

And here's my current settings.json
```json
{
    "workbench.editorAssociations": {
        "{git,gitlens}:/**/*.{md,csv,svg}": "default"
    },
    "editor.minimap.enabled": false,
    "breadcrumbs.enabled": false,
    "workbench.startupEditor": "newUntitledFile",
    "window.restoreWindows": "none",
    "workbench.editor.untitled.labelFormat": "name",
    "workbench.editor.restoreViewState": false,
    "files.hotExit": "off",
    "window.closeWhenEmpty": true,
    "workbench.activityBar.location": "hidden",
}
```

Expectation: VSCode will open without 'explorer' sidebar
Reality: VSCode will open with an 'explorer' sidebar
Reproduction: Close vscode entirely. Open vscode, witness that it opens with an explorer sidetab. 

I've also tried all associated settings I could find like `"explorer.autoReveal": false,` but they had no effect on mitigating this issue.

VS Code version: Code 1.94.0 (d78a74bcdfad14d5d3b1b782f87255d802b57511, 2024-10-02T13:08:12.626Z)
OS version: Windows_NT x64 10.0.19045
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i9-10900K CPU @ 3.70GHz (20 x 3696)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled<br>webnn: disabled_off|
|Load (avg)|undefined|
|Memory (System)|127.88GB (112.07GB free)|
|Process Argv|--crash-reporter-id 6cf1cdf9-0be8-4dac-b5a5-b9e955d4dbd9|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (23)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-office|cwe|3.4.3
dart-code|Dar|3.98.0
flutter|Dar|3.98.0
gc-excelviewer|Gra|4.2.62
blender-development|Jac|0.0.24
csdevkit|ms-|1.10.18
csharp|ms-|2.45.25
vscode-dotnet-runtime|ms-|2.1.7
debugpy|ms-|2024.10.0
isort|ms-|2023.10.1
python|ms-|2024.16.0
vscode-pylance|ms-|2024.9.2
jupyter|ms-|2024.9.1
jupyter-keymap|ms-|1.1.2
jupyter-renderers|ms-|1.0.19
vscode-jupyter-cell-tags|ms-|0.1.9
vscode-jupyter-slideshow|ms-|0.1.6
remote-wsl|ms-|0.88.4
cpptools|ms-|1.21.6
vscode-yaml|red|1.15.0
cmake|twx|0.0.17
intellicode-api-usage-examples|Vis|0.2.8
vscodeintellicode|Vis|1.3.1

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vspor879:30202332
vspor708:30202333
vspor363:30204092
vswsl492cf:30256860
vscod805cf:30301675
binariesv615:30325510
vsaa593:30376534
py29gd2263:31024239
c4g48928:30535728
azure-dev_surveyone:30548225
a9j8j154:30646983
962ge761:30959799
pythongtdpath:30769146
welcomedialog:30910333
pythonnoceb:30805159
asynctok:30898717
pythonmypyd1:30879173
h48ei257:31000450
pythontbext0:30879054
accentitlementst:30995554
dsvsc016:30899300
dsvsc017:30899301
dsvsc018:30899302
cppperfnew:31000557
dsvsc020:30976470
pythonait:31006305
dsvsc021:30996838
0ee40948:31013168
a69g1124:31058053
dvdeprecation:31068756
dwnewjupytercf:31046870
impr_priority:31102340
nativerepl1:31139838
refactort:31108082
pythonrstrctxt:31112756
flighttreat:31134774
wkspc-onlycs-t:31132770
wkspc-ranged-t:31151552
cf971741:31144450
autoexpandse:31146404
iacca2:31150323
notype1:31151523
cc771715:31146322

```

</details>

**Comment by andmm** [comment]
seems related to my issue: https://github.com/microsoft/vscode/issues/230489

**Comment by silverqx** [comment]
This is very annoying issue without any workaround that takes 25% of the edit area for no reason.

I'm using vscode for editing everything like random txt, ini, json files and in 95% of cases I have to close this sidebar as it only contains Open Folder and Clone Repository buttons.
For me it's also annoying for workspaces as I'm using this sidebar very rare because I'm using breadcrumb navigation in most cases.

Also for me when I open vscode then it restores ~5 vscode windows because I have opened more workspaces at once like dotfiles, current project, another random txt file, or ahk workspace and it everywhere opens this sidebar, very annoying for geek users.

The best solution would be to save it to the state per workspace and also add a setting to control this at startup for new vscode windows, or at least restore the previous behavior. I hope this will not be the new standard like a regression in search #206511

**Comment by samcarter** [comment]
@silverqx as a temporary workaround, maybe have a look at https://marketplace.visualstudio.com/items?itemName=sirmspencer.vscode-autohide . The sidebar will still be open for a moment, but then gets hidden automatically.

**Comment by silverqx** [comment]
Thx @samcarter, I know about this extension but I don't like it either, the auto-hide when you click on the edit area is also weird, I might give it a shot, I will hope that this will be fixed soon.

**Comment by benibenj** [comment]
Thanks for reporting, I will be reverting the change which is causing this to happen.

**Comment by silverqx** [comment]
Thx for asap fix, I'm happy about it. 👍

**Comment by Sawtaytoes** [comment]
Thank you so much for reverting! Can't wait for the fix :). Been bugging me for weeks!

I just got frustrated enough to file an issue, but it looks like multiple others did for me and even fixed the issue ;).

**Comment by alexruimy** [comment]
This is still happening for me on 1.95.0.

I think it's less about the sidebar than it is about Explorer. I say this because sometimes when I have the sidebar open for Search, it will jump to Explorer for no apparent reason. If the sidebar is closed, it will open as described in this thread.

**Comment by Sawtaytoes** [comment]
I can confirm this is fixed in Windows 11 64-bit 23H2 in `1.95`. I had the sidebar closed, and when I opened up VS Code after updating, the sidebar was still closed.

I tried this a few times, and the results are the same. Fantastic! Thanks for this fix!

**Comment by silverqx** [comment]
It's okay in v1.95. The sidebar open state is even persisted, which is great; I like it. 🕺
